modalc-lib/modal-vector.rkt

```diff
@@ -0,0 +1,45 @@
+#lang racket
+
+(require modalc)
+
+(define (modal-vector/c should-apply-ctc? . inner-ctc)
+  (define vector-contract (apply vector/c inner-ctc))
+  (define inner-ctc-proj
+    (contract-late-neg-projection vector-contract))
```

Comment:
We don't want to do this, because this is creating the entire vector contract -- whereas `modal-vector/c` should be a replacement for the vector part of the contract.

Reply:
Instead we need to create projections for all the inner contracts.

Reply:
For reference about doing this and also checks below inside the impersonator, check out lines 50, 63, 79, and 80 of `modal.rkt`, in the implementation of `modal->`